src/replication/replication_node.cpp

```diff
@@ -34,10 +34,10 @@ namespace cubreplication
 {
   replication_node::~replication_node ()
   {
-    delete m_stream;
-    m_stream = NULL;
     delete m_stream_file;
     m_stream_file = NULL;
+    delete m_stream;
```

Comment:
It seems stream and stream file are interdependent. If stream file is destroyed first, stream's m_filled_stream_handler becomes invalid (its second argument is reference to stream file).

Reply:
fixed: by stopping first the stream and checking for stopped flag before calling m_filled_stream_handler.
I'll wait also for @razvanradulescu's input regarding this changes

Reply:
For a quick fix, I think this is enough.
The bigger issue is the classes are interdependent; after will become more clear which functionalities are required (of managing replication files), we may slightly improve the design.